src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -205,22 +205,18 @@ public int read() throws IOException {
 
     private int traceRead0() throws IOException {
         int result = 0;
-        boolean endOfFile = false;
         long bytesRead = 0;
         long start = 0;
         try {
             start = FileReadEvent.timestamp();
             result = read0();
             if (result < 0) {
-                endOfFile = true;
+                bytesRead = -1;
             } else {
                 bytesRead = 1;
             }
         } finally {
-            long duration = FileReadEvent.timestamp() - start;
-            if (FileReadEvent.shouldCommit(duration)) {
-                FileReadEvent.commit(start, duration, path, bytesRead, endOfFile);
-            }
+            FileReadEvent.offer(start, path, bytesRead);
```

Comment:
Thanks for this update, this looks much better. It think would be better again if `start = FileReadEvent.timestamp()` is moved to before the try.